presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -831,6 +795,29 @@ else if (type instanceof MapType) {
         return new RelationPlan(unnestNode, scope, unnestedSymbols);
     }
 
+    private void computeUnnestMappings(ImmutableMap.Builder<Symbol, List<Symbol>> mappings, Iterator<Symbol> outputs, Symbol input, Type type)
```

Comment:
- I don't like Iterator being an argument here.  I now it's preexisting, but extracting method makes it more apparent. I am concerned some day an off by one problem will creep here.

- you can make the method return `Map<..>` (or `Entry`) instead of passing `ImmutableMap.Builder`.

Reply:
I know. The problem is that the iterator and the map builder are "stateful" across multiple invocations of computeUnnestMappings. Let me see if I can fix that some way.

Reply:
ideally we would record the during analysis, instead of matching (semi-)positionally unnest symbols with output descriptor.

Reply:
Yup, that’s a bigger cleanup I’ve been working on, but didn’t want to complicate this one too much.